.circleci/config.yml

```diff
@@ -23,120 +23,29 @@ version: 2
-            
-            cd /home/V2X-Hub-build/
-            mkdir -p ext
-            cd /home/V2X-Hub-build/ext/
-            git clone https://github.com/usdot-fhwa-OPS/libwebsockets.git
-            cd /home/V2X-Hub-build/ext/libwebsockets/
-            cmake -DLWS_WITH_SHARED=OFF .
-            make
-            make install
-            cd ../../../..
-
-            cd /home/V2X-Hub-build/ext
-            git clone https://github.com/usdot-fhwa-OPS/qhttpengine.git
-            cd /home/V2X-Hub-build/ext/qhttpengine
-            cmake .
-            make 
-            make install
-            
-            cd /home/V2X-Hub-build/ext/server
-            cmake .
-            make
-            make install
-            
-            cd /usr/src/googletest
-            mkdir build
-            cd /usr/src/googletest/build
-            cmake ..
-            make
-            make install
-            cd /usr/src/googletest
-            rm -rf build
-            ln -s  /usr/local/lib/libgtest.a /usr/lib/libgtest.a 
-            ln -s  /usr/local/lib/libgtest_main.a /usr/lib/libgtest_main.a
-            cd /usr/local/lib/
-            ldconfig
-
-            cd ../../../../..
-            cd /home/V2X-Hub-build
-            mkdir .base-image
-            SONAR_DIR=/opt/sonarqube
-            mkdir $SONAR_DIR
-            curl -o $SONAR_DIR/sonar-scanner.zip https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.2.0.1873-linux.zip
-            curl -o $SONAR_DIR/build-wrapper.zip https://sonarcloud.io/static/cpp/build-wrapper-linux-x86.zip
-            curl -sL https://deb.nodesource.com/setup_10.x | bash -
-            sudo apt-get install -y nodejs
-            mkdir /opt/jq
-            curl -L "https://github.com/stedolan/jq/releases/download/jq-1.5/jq-linux64" -o /opt/jq/jq
-            chmod +x /opt/jq/jq
-            cd $SONAR_DIR
-            unzip $SONAR_DIR/sonar-scanner.zip -d .
-            unzip $SONAR_DIR/build-wrapper.zip -d .
-            rm $SONAR_DIR/sonar-scanner.zip
-            rm $SONAR_DIR/build-wrapper.zip
-            mv $(ls $SONAR_DIR | grep "sonar-scanner-") $SONAR_DIR/sonar-scanner/
-            mv $(ls $SONAR_DIR | grep "build-wrapper-") $SONAR_DIR/build-wrapper/
-            echo "export PATH=$PATH:/opt/jq/:$SONAR_DIR/sonar-scanner/bin/:$SONAR_DIR/build-wrapper/" > /home/V2X-Hub-build/.base-image/init-env.sh
-      - run:
-          name: Build V2I-Hub
-          command: |
-            set -x
-            source /home/V2X-Hub-build/.base-image/init-env.sh
-            cd /home/V2X-Hub-build/src
-            build-wrapper-linux-x86-64 --out-dir ../bw-output ./build.sh
-      - run:
-          name: Run Tests & Generate Gcovr Reports
-          command: |
-            set -x
-            cd /home/V2X-Hub-build/src
-            ./coverage.sh
-      - run:
-          name: Run Sonar Scanner
```

Comment:
Might be necessary to keep this step since it is not replicated in the dockerfile. Also will investigate whether the V2XHub code needs to be compiled to be scanned by the sonar scanner tool.

Reply:
Added the old script back as well and will write an issue to investigate Sonar-scanner tool separately for Dockerfile and CircleCI script

Reply:
@paulbourelly999  I do not think we need a docker push to  push the image to dockerhub at the circleCI config. The dockerhub itself will automatically pickup the changes from a configured github branch.